tests/beta/tapes/test_tape.py

```diff
@@ -0,0 +1,667 @@
+        """Test that free parameters are correctly modified after construction"""
+        tape, params = make_tape
+        new_params = [-0.654, 0.3]
+
+        tape.trainable_params = {1, 3}
+        tape.set_parameters(new_params)
+
+        count = 0
+        for idx, pinfo in tape._par_info.items():
+            if idx in tape.trainable_params:
+                assert pinfo["op"].data[pinfo["p_idx"]] == new_params[count]
+                count += 1
+            else:
+                assert pinfo["op"].data[pinfo["p_idx"]] == params[idx]
+
+        assert tape.get_parameters(free_only=False) == [
+            params[0],
+            new_params[0],
+            params[2],
+            new_params[1],
+            params[4],
+        ]
+
+    def test_setting_all_parameters(self, make_tape):
+        """Test that all parameters are correctly modified after construction"""
+        tape, params = make_tape
+        new_params = [0.6543, -0.654, 0, 0.3, 0.6]
+
+        tape.trainable_params = {1, 3}
+        tape.set_parameters(new_params, free_only=False)
+
+        for pinfo, pval in zip(tape._par_info.values(), new_params):
+            assert pinfo["op"].data[pinfo["p_idx"]] == pval
+
+        assert tape.get_parameters(free_only=False) == new_params
+
+    def test_setting_parameters_error(self, make_tape):
+        """Test that exceptions are raised if incorrect parameters
+        are attempted to be set"""
+        tape, _ = make_tape
+
+        with pytest.raises(ValueError, match="Number of provided parameters invalid"):
+            tape.set_parameters([0.54])
+
+        with pytest.raises(ValueError, match="Number of provided parameters invalid"):
+            tape.trainable_params = {2, 3}
+            tape.set_parameters([0.54, 0.54, 0.123])
+
+    def test_array_parameter(self):
+        """Test that array parameters integrate properly"""
+        a = np.array([1, 1, 0, 0]) / np.sqrt(2)
+        params = [a, 0.32, 0.76, 1.0]
+
+        with QuantumTape() as tape:
+            op = qml.QubitStateVector(params[0], wires=0)
+            qml.Rot(params[1], params[2], params[3], wires=0)
+
+        assert tape.num_params == len(params)
+        assert tape.get_parameters() == params
+
+        b = np.array([0, 1, 0, 0])
+        new_params = [b, 0.543, 0.654, 0.123]
+        tape.set_parameters(new_params)
+        assert tape.get_parameters() == new_params
+
+        assert np.all(op.data[0] == b)
+
+    def test_measurement_parameter(self):
+        """Test that measurement parameters integrate properly"""
+        H = np.array([[1, 0], [0, -1]])
+        params = [0.32, 0.76, 1.0, H]
+
+        with QuantumTape() as tape:
+            qml.Rot(params[0], params[1], params[2], wires=0)
+            obs = qml.Hermitian(params[3], wires=0)
+            expval(obs)
+
+        assert tape.num_params == len(params)
+        assert tape.get_parameters() == params
+
+        H2 = np.array([[0, 1], [1, 1]])
```

Comment:
lol I guess this is a sneaky way to bypass any checks in `qml.Hermitian` that the matrix is actually Hermitian? :laughing:

Reply:
I _think_ the checks actually happen on execution only, hence why this doesn't cause any issue!